hibernate-cache/src/main/java/org/infinispan/hibernate/cache/access/LockingInterceptor.java

```diff
@@ -34,17 +34,14 @@
 public class LockingInterceptor extends NonTransactionalLockingInterceptor {
 	private static final Log log = LogFactory.getLog(LockingInterceptor.class);
 
-   protected final InvocationFinallyAction unlockAllReturnCheckCompletableFutureHandler = new InvocationFinallyAction() {
-      @Override
-      public void accept(InvocationContext rCtx, VisitableCommand rCommand, Object rv, Throwable throwable) throws Throwable {
-         lockManager.unlockAll(rCtx);
-         if (rv instanceof CompletableFuture) {
-            try {
-               ((CompletableFuture) rv).join();
-            }
-            catch (CompletionException e) {
-               throw e.getCause();
-            }
+   private final InvocationFinallyAction unlockAllReturnCheckCompletableFutureHandler = (rCtx, rCommand, rv, throwable) -> {
+      lockManager.unlockAll(rCtx);
+      if (rv instanceof CompletableFuture) {
```

Comment:
Need a comment, when is `rv` a `CompletableFuture`?

Reply:
hibernate code. I give up on trying to understanding it.

Reply:
@danberindei The CF is produced in UnorderedDistributionInterceptor, IIRC we want the EWI to commit the entry & unlock before the remote call completes but at some point we need to make the call blocking, waiting for the other party.
I'll add a code comment.